### PR TITLE
add grpc metrics to telemetry

### PR DESCRIPTION
**Pull request by vangheem**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

**Comment by vangheem** [comment]
[ch-4850]

**Review by bloodbare — COMMENTED** [review]
Can we keep the old names ? this library is used a lot on all the platform.

File: nucliadb_telemetry/nucliadb_telemetry/grpc_metrics.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+class StreamUnaryClientInterceptor(aio.StreamUnaryClientInterceptor):
+    async def intercept_stream_unary(
+        self, continuation, client_call_details: ClientCallDetails, request_iterator
+    ):
+        metric = grpc_client_observer(labels={"method": client_call_details.method})
+        call = await continuation(client_call_details, request_iterator)
+        call.add_done_callback(functools.partial(finish_metric_grpc, metric))
+
+        return call
```
Comment (bloodbare) [inline nucliadb_telemetry/nucliadb_telemetry/grpc_metrics.py:95]: We also have UnaryStream GRPC services

Comment (vangheem) [reply nucliadb_telemetry/nucliadb_telemetry/grpc_metrics.py:156]: Okay, after thinking it over again, I figure we should just observe these as normal and if any additional custom instrumentation needs to be done, that can be done on a case-by-case basic.

**Pull request merged**